src/Orleans/CodeGeneration/TypeUtils.cs

```diff
@@ -512,14 +512,41 @@ public static bool TryResolveType(string fullName, out Type type)
             return CachedTypeResolver.Instance.TryResolveType(fullName, out type);
         }
 
+        private static Lazy<bool> canUseReflectionOnly = new Lazy<bool>(() =>
+        {
+            try
+            {
+                CachedReflectionOnlyTypeResolver.Instance.TryResolveType(typeof(TypeUtils).AssemblyQualifiedName, out _);
+                return true;
+            }
+            catch (PlatformNotSupportedException)
+            {
+                return false;
+            }
+            catch (Exception)
+            {
+                // if other exceptions not related to platform ocurr, assume that ReflectionOnly is supported
```

Comment:
wouldn't false be a safer option?

Reply:
well, I added that as I was troubleshooting the other issue (where I unknowingly included a change from the previous PR, and was getting some loaded assemblies when I was expecting them to be reflection only load).
The reality is that I don't expect that condition to ever be hit (and I never saw it being hit even while troubleshooting the other error). But still, in the absolutely unlikely situation that it does occur, then getting a PlatformNotSupportedException immediately is the only thing that should turn this feature off. Other non-related failures should not turn off reflection only. So even though unlikely, the logic IMO is correct.